inc/project.class.php

```diff
@@ -237,6 +237,25 @@ static function getAdditionalMenuLinks() {
    function post_updateItem($history = 1) {
       global $CFG_GLPI;
 
+      $project = new Project();
+      if ($this->fields['auto_percent_done']) {
+         // Force recalculation of this and parents
+         $this->recalculatePercentDone();
```

Comment:
I have only one project in my test instance, and it loops here. Indeed, `recalculatePercentDone` calls `update` which calls `post_updateItem` and so on.

To prevent this behaviour, I propose to :
1. Have a `recalculatePercentDone` method that only return the new value (without trigerring itself the update);
2. Update the field in `preprareInputForUpdate` (`$input['percent_done'] = $this->recalculatePercentDone();`), but only when `auto_percent_done` field value changed;
3. Update the parent with new computed value in `post_updateItem` (`$parent->update(['id' => $parent->getID(), 'percent_done' => $parent->recalculatePercentDone()])`), but only if percent_done value changed for self;

I am not sure that will work, need to be tested.

Reply:
I will need to revert some of the last round of changes, test on a clean install, and create unit tests. Recursion on the calculation should not be possible. Function triggers the parent to update only (after updating its percent done) which recalculates for its parent and so on.

Reply:
I made a mistake in one of the later commits (used fields instead of input). The loop should be fixed now.